src/ejabberd_c2s.erl

```diff
@@ -3268,6 +3296,7 @@ handover_session(SD) ->
                               resumed),
     {N, Messages} = flush_messages(),
     NewSize = N + SD#state.stream_mgmt_buffer_size,
+    notify_unacknowledged_messages(SD,Messages),
```

Comment:
These messages are going to be delivered to the new session. Why should we push them?

Reply:
the idea is to push notifications as fast as possible after MIM suggests the client is disconnected. Android client team said that it's the best practice and this behaviour is "good to have" right now but it will become "critical" in the next versions of Android because of more aggressive app hibernation policies to improve energy efficiency. also I've been told that Apple is thinking about pretty common restrictions for the future version of iOS.

Reply:
:+1: